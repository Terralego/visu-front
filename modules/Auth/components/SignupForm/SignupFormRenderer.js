import React from 'react';
import { Card, FormGroup, InputGroup, Button, Intent } from '@blueprintjs/core';

export const SignupFormRenderer = ({
  submit,
  setSignupProperty,
  errors: { email, password } = {},
}) => (
  <Card className="signup-form">
    <form
      onSubmit={submit}
    >
      <FormGroup
        helperText="Type your email"
        label="Email"
        labelFor="email"
        labelInfo="(required)"
        intent={email ? Intent.WARNING : null}
      >
        <InputGroup
          id="email"
          placeholder="Email"
          onChange={setSignupProperty}
          intent={email ? Intent.WARNING : null}
          autoComplete="username"
        />
      </FormGroup>

      <FormGroup
        helperText="Type your password"
        label="Password"
        labelFor="password"
        labelInfo="(required)"
        intent={password ? Intent.WARNING : null}
      >
        <InputGroup
          type="password"
          id="password"
          placeholder="Password"
          onChange={setSignupProperty}
          intent={password ? Intent.WARNING : null}
          autoComplete="current-password"
        />
      </FormGroup>

      <Button
        type="submit"
      >
        Inscription
      </Button>
    </form>
  </Card>
);


export default SignupFormRenderer;
