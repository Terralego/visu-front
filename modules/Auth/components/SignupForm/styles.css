.signup-form {
  max-width: 40rem;
  margin: auto;
}