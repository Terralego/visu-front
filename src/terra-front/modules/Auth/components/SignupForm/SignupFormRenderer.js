import React from 'react';
import { Card, FormGroup, InputGroup, Button, Intent, Callout } from '@blueprintjs/core';

import translateMock from '../../../../utils/translate';

export const SignupFormRenderer = ({
  submit,
  setSignupProperty,
  errors: { email, password } = {},
  showPassword,
  loading,
  done,
  sentMail,
  translate = translateMock({
    'auth.signupform.email.invalid': 'Invalid email',
    'auth.signupform.email.help': 'Type your email',
    'auth.signupform.email.label': 'Email',
    'auth.signupform.email.info': 'required',
    'auth.signupform.email.placeholder': 'Email',
    'auth.signupform.password.invalid': 'Password is required',
    'auth.signupform.password.help': 'Type your password',
    'auth.signupform.password.label': 'Password',
    'auth.signupform.password.info': 'required',
    'auth.signupform.password.placeholder': 'Password',
    'auth.signupform.submit': 'Create account',
    'auth.signupform.submitSuccess': 'Email sent to {{email}}',
  }),
}) => (
  <Card className="signup-form">
    {done && (
      <Callout className="signup-form__callout" intent={Intent.SUCCESS}>
        {translate('auth.signupform.submitSuccess', { email: sentMail })}
      </Callout>
    )}
    <form
      onSubmit={submit}
    >
      <FormGroup
        helperText={translate('auth.signupform.email.help')}
        label={translate('auth.signupform.email.label')}
        labelFor="email"
        labelInfo={translate('auth.signupform.email.info')}
        intent={email ? Intent.WARNING : null}
      >
        <InputGroup
          type="email"
          id="email"
          placeholder={translate('auth.signupform.email.placeholder')}
          onChange={setSignupProperty}
          intent={email ? Intent.WARNING : null}
          autoComplete="username"
        />
      </FormGroup>

      {showPassword && (
        <FormGroup
          helperText={translate('auth.signupform.password.help')}
          label={translate('auth.signupform.password.label')}
          labelFor="password"
          labelInfo={translate('auth.signupform.password.info')}
          intent={password ? Intent.WARNING : null}
        >
          <InputGroup
            type="password"
            id="password"
            placeholder={translate('auth.signupform.password.placeholder')}
            onChange={setSignupProperty}
            intent={password ? Intent.WARNING : null}
            autoComplete="current-password"
          />
        </FormGroup>
      )}

      <Button
        type="submit"
        disabled={loading}
      >
        {translate('auth.signupform.submit')}
      </Button>
    </form>
  </Card>
);


export default SignupFormRenderer;
