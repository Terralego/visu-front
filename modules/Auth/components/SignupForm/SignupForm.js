import React, { Component } from 'react';
import PropTypes from 'prop-types';

import render from './SignupFormRenderer';

import './styles.css';

export class SignupForm extends Component {
  static propTypes = {
    render: PropTypes.func,
  };

  static defaultProps = {
    render,
  };

  state = {};

  setSignupProperty = ({ target: { id, value } }) => this.setState({ [id]: value });

  submit = async event => {
    event.preventDefault();
    const properties = { ...this.state };
    this.setState({
      errors: {
        ...Object.keys(this.state).reduce((acc, curr) => ({ ...acc, [curr]: false }), {}),
      },
    });

    const { signupAction } = this.props;

    try {
      await signupAction(properties);
    } catch (error) {
      if (error.data) {
        this.setState({
          errors: {
            email: !!error.data.email,
            password: !!error.data.password,
          },
        });
      }
    }
  }

  render () {
    const { render: Render } = this.props;
    const { errors } = this.state;
    const { setSignupProperty, submit } = this;

    const props = {
      setSignupProperty, submit, errors,
    };

    return (
      <Render {...props} />
    );
  }
}

export default SignupForm;
