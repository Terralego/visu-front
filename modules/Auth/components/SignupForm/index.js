import SignupForm from './SignupForm';
import { connectTerraFrontProvider } from '../../../TerraFrontProvider';
import { connectAuthProvider } from '../../services/context';

export default connectTerraFrontProvider(({
  components: { SignupForm: SignupFormProps = {} } = {},
}) => SignupFormProps)(connectAuthProvider(({
  signupAction,
}) => ({ signupAction }))(SignupForm));
