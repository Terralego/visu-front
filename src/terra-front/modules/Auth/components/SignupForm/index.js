import compose from '../../../../utils/compose';
import SignupForm from './SignupForm';
import { connectTerraFrontProvider } from '../../../TerraFrontProvider';
import { connectAuthProvider } from '../../services/context';

export default compose(
  connectTerraFrontProvider({ '*': 'modules.Auth.components.SignupForm' }),
  connectAuthProvider(({ signupAction }) => ({ signupAction })),
)(SignupForm);
