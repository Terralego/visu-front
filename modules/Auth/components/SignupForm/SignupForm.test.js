import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import SignupForm from './SignupForm';

it('should render correctly', () => {
  const tree = renderer
    .create(<SignupForm />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('should render with errors', () => {
  const wrapper = renderer
    .create(<SignupForm />);
  wrapper.getInstance().setState({ errors: { email: true } });
  const tree = wrapper.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should submit form with success', async done => {
  const signupAction = jest.fn();
  const wrapper = shallow(<SignupForm
    signupAction={signupAction}
  />);

  wrapper.setState({ email: 'foo@bar', password: 'bar', confirmEmail: 'foo@bar', confirmPassword: 'bar' });

  await wrapper.instance().submit({ preventDefault () {} });

  expect(signupAction).toHaveBeenCalledWith({
    email: 'foo@bar',
    password: 'bar',
    confirmEmail: 'foo@bar',
    confirmPassword: 'bar',
  });
  expect(wrapper.state().errors.email).toBe(false);
  expect(wrapper.state().errors.password).toBe(false);
  expect(wrapper.state().errors.confirmEmail).toBe(false);
  expect(wrapper.state().errors.confirmPassword).toBe(false);

  done();
});

it('should submit form with failure', async done => {
  const signupAction = jest.fn(() => {
    const error = new Error();
    error.data = {
      email: ['invalid'],
    };
    throw error;
  });
  const wrapper = shallow(<SignupForm
    signupAction={signupAction}
  />);

  await wrapper.instance().submit({ preventDefault () {} });

  expect(signupAction).toHaveBeenCalledWith({ email: undefined, password: undefined });
  expect(wrapper.state().errors.email).toBe(true);
  expect(wrapper.state().errors.password).toBe(false);

  done();
});

it('should set property', () => {
  const wrapper = shallow(<SignupForm />);
  wrapper.instance().setSignupProperty({ target: { value: 'foo', id: 'email' } });
  wrapper.instance().setSignupProperty({ target: { value: 'foo', id: 'password' } });
  expect(wrapper.state().email).toBe('foo');
  expect(wrapper.state().password).toBe('foo');
});
