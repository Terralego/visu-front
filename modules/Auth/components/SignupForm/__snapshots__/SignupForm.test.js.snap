// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="bp3-card bp3-elevation-0 signup-form"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="bp3-form-group"
    >
      <label
        className="bp3-label"
        htmlFor="email"
      >
        Email
         
        <span
          className="bp3-text-muted"
        >
          (required)
        </span>
      </label>
      <div
        className="bp3-form-content"
      >
        <div
          className="bp3-input-group"
        >
          <input
            autoComplete="username"
            className="bp3-input"
            id="email"
            onChange={[Function]}
            placeholder="Email"
            style={
              Object {
                "paddingRight": 10,
              }
            }
            type="text"
          />
        </div>
        <div
          className="bp3-form-helper-text"
        >
          Type your email
        </div>
      </div>
    </div>
    <div
      className="bp3-form-group"
    >
      <label
        className="bp3-label"
        htmlFor="confirmEmail"
      >
        Confirm your email
         
        <span
          className="bp3-text-muted"
        >
          (required)
        </span>
      </label>
      <div
        className="bp3-form-content"
      >
        <div
          className="bp3-input-group"
        >
          <input
            autoComplete="email"
            className="bp3-input"
            id="confirmEmail"
            onChange={[Function]}
            placeholder="Confirm your email"
            style={
              Object {
                "paddingRight": 10,
              }
            }
            type="text"
          />
        </div>
        <div
          className="bp3-form-helper-text"
        >
          Confirm your email
        </div>
      </div>
    </div>
    <div
      className="bp3-form-group"
    >
      <label
        className="bp3-label"
        htmlFor="password"
      >
        Password
         
        <span
          className="bp3-text-muted"
        >
          (required)
        </span>
      </label>
      <div
        className="bp3-form-content"
      >
        <div
          className="bp3-input-group"
        >
          <input
            autoComplete="current-password"
            className="bp3-input"
            id="password"
            onChange={[Function]}
            placeholder="Password"
            style={
              Object {
                "paddingRight": 10,
              }
            }
            type="password"
          />
        </div>
        <div
          className="bp3-form-helper-text"
        >
          Type your password
        </div>
      </div>
    </div>
    <div
      className="bp3-form-group"
    >
      <label
        className="bp3-label"
        htmlFor="confirmPassword"
      >
        Confirm your password
         
        <span
          className="bp3-text-muted"
        >
          (required)
        </span>
      </label>
      <div
        className="bp3-form-content"
      >
        <div
          className="bp3-input-group"
        >
          <input
            autoComplete="current-password"
            className="bp3-input"
            id="confirmPassword"
            onChange={[Function]}
            placeholder="Confirm your Password"
            style={
              Object {
                "paddingRight": 10,
              }
            }
            type="password"
          />
        </div>
        <div
          className="bp3-form-helper-text"
        >
          Confirm your password
        </div>
      </div>
    </div>
    <button
      className="bp3-button"
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      type="submit"
    >
      <span
        className="bp3-button-text"
      >
        Inscription
      </span>
    </button>
  </form>
</div>
`;

exports[`should render with errors 1`] = `
<div
  className="bp3-card bp3-elevation-0 signup-form"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="bp3-form-group"
    >
      <label
        className="bp3-label"
        htmlFor="email"
      >
        Email
         
        <span
          className="bp3-text-muted"
        >
          (required)
        </span>
      </label>
      <div
        className="bp3-form-content"
      >
        <div
          className="bp3-input-group"
        >
          <input
            autoComplete="username"
            className="bp3-input"
            id="email"
            onChange={[Function]}
            placeholder="Email"
            style={
              Object {
                "paddingRight": 10,
              }
            }
            type="text"
          />
        </div>
        <div
          className="bp3-form-helper-text"
        >
          Type your email
        </div>
      </div>
    </div>
    <div
      className="bp3-form-group"
    >
      <label
        className="bp3-label"
        htmlFor="confirmEmail"
      >
        Confirm your email
         
        <span
          className="bp3-text-muted"
        >
          (required)
        </span>
      </label>
      <div
        className="bp3-form-content"
      >
        <div
          className="bp3-input-group"
        >
          <input
            autoComplete="email"
            className="bp3-input"
            id="confirmEmail"
            onChange={[Function]}
            placeholder="Confirm your email"
            style={
              Object {
                "paddingRight": 10,
              }
            }
            type="text"
          />
        </div>
        <div
          className="bp3-form-helper-text"
        >
          Confirm your email
        </div>
      </div>
    </div>
    <div
      className="bp3-form-group"
    >
      <label
        className="bp3-label"
        htmlFor="password"
      >
        Password
         
        <span
          className="bp3-text-muted"
        >
          (required)
        </span>
      </label>
      <div
        className="bp3-form-content"
      >
        <div
          className="bp3-input-group"
        >
          <input
            autoComplete="current-password"
            className="bp3-input"
            id="password"
            onChange={[Function]}
            placeholder="Password"
            style={
              Object {
                "paddingRight": 10,
              }
            }
            type="password"
          />
        </div>
        <div
          className="bp3-form-helper-text"
        >
          Type your password
        </div>
      </div>
    </div>
    <div
      className="bp3-form-group"
    >
      <label
        className="bp3-label"
        htmlFor="confirmPassword"
      >
        Confirm your password
         
        <span
          className="bp3-text-muted"
        >
          (required)
        </span>
      </label>
      <div
        className="bp3-form-content"
      >
        <div
          className="bp3-input-group"
        >
          <input
            autoComplete="current-password"
            className="bp3-input"
            id="confirmPassword"
            onChange={[Function]}
            placeholder="Confirm your Password"
            style={
              Object {
                "paddingRight": 10,
              }
            }
            type="password"
          />
        </div>
        <div
          className="bp3-form-helper-text"
        >
          Confirm your password
        </div>
      </div>
    </div>
    <button
      className="bp3-button"
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      type="submit"
    >
      <span
        className="bp3-button-text"
      >
        Inscription
      </span>
    </button>
  </form>
</div>
`;
